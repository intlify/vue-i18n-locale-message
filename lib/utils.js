"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const component_compiler_utils_1 = require("@vue/component-compiler-utils");
const compiler = __importStar(require("vue-template-compiler"));
const fs_1 = __importDefault(require("fs"));
const glob_1 = __importDefault(require("glob"));
const path_1 = __importDefault(require("path"));
const json5_1 = __importDefault(require("json5"));
const js_yaml_1 = __importDefault(require("js-yaml"));
const debug_1 = require("debug");
const debug = debug_1.debug('vue-i18n-locale-message:utils');
const ESC = {
    '<': '&lt;',
    '>': '&gt;',
    '"': '&quot;',
    '&': '&amp;'
};
function escape(s) {
    return s.replace(/[<>"&]/g, escapeChar);
}
exports.escape = escape;
function escapeChar(a) {
    return ESC[a] || a;
}
function resolve(...paths) {
    return path_1.default.resolve(...paths);
}
exports.resolve = resolve;
function reflectSFCDescriptor(basePath, components) {
    return components.map(target => {
        const { template, script, styles, customBlocks } = component_compiler_utils_1.parse({
            source: target.content,
            filename: target.path,
            compiler: compiler
        });
        return {
            ...parsePath(basePath, target.path),
            raw: target.content,
            customBlocks,
            template,
            script,
            styles
        };
    });
}
exports.reflectSFCDescriptor = reflectSFCDescriptor;
function parsePath(basePath, targetPath) {
    const { dir, name } = path_1.default.parse(targetPath);
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    const [_, target] = dir.split(basePath);
    const parsedTargetPath = target.split(path_1.default.sep);
    parsedTargetPath.shift();
    debug(`parsePath: contentPath = ${targetPath}, component = ${name}, messageHierarchy = ${parsedTargetPath}`);
    return {
        contentPath: targetPath,
        component: name,
        hierarchy: [...parsedTargetPath, name]
    };
}
exports.parsePath = parsePath;
function parseContent(content, lang) {
    switch (lang) {
        case 'yaml':
        case 'yml':
            return js_yaml_1.default.safeLoad(content);
        case 'json5':
            return json5_1.default.parse(content);
        case 'json':
        default:
            return JSON.parse(content);
    }
}
exports.parseContent = parseContent;
function stringifyContent(content, lang, options) {
    var _a, _b;
    const indent = ((_a = options) === null || _a === void 0 ? void 0 : _a.intend) || 2;
    const eof = ((_b = options) === null || _b === void 0 ? void 0 : _b.eof) || '\n';
    let result = '';
    switch (lang) {
        case 'yaml':
        case 'yml':
            result = js_yaml_1.default.safeDump(content, { indent });
            break;
        case 'json5':
            result = json5_1.default.stringify(content, null, indent);
            break;
        case 'json':
        default:
            result = JSON.stringify(content, null, indent);
            break;
    }
    if (!result.endsWith(eof)) {
        result += eof;
    }
    return result;
}
exports.stringifyContent = stringifyContent;
function readSFC(target) {
    const targets = resolveGlob(target);
    debug('readSFC: targets = ', targets);
    // TODO: async implementation
    return targets.map(target => {
        const data = fs_1.default.readFileSync(target);
        return {
            path: target,
            content: data.toString()
        };
    });
}
exports.readSFC = readSFC;
function resolveGlob(target) {
    // TODO: async implementation
    return glob_1.default.sync(`${target}/**/*.vue`);
}
