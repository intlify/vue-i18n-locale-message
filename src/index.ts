import squeeze from './squeezer'
import infuse from './infuser'
import { getTranslationStatus as status } from './utils'

export {
  squeeze,
  infuse,
  status
}
