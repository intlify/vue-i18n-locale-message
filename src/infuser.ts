/* eslint-disable-next-line */
/// <reference path="../types/shims-vue-template-compiler.d.ts"/>
import { SFCBlock, SFCDescriptor } from 'vue-template-compiler'
import { Locale, MetaLocaleMessage, SFCI18nBlock, SFCFileInfo, FormatOptions } from '../types'

import { escape, reflectSFCDescriptor, parseContent, stringifyContent } from './utils'

import { debug as Debug } from 'debug'
const debug = Debug('vue-i18n-locale-message:infuser')

export default function infuse (basePath: string, sources: SFCFileInfo[], meta: MetaLocaleMessage, options?: FormatOptions): SFCFileInfo[] {
  const descriptors = reflectSFCDescriptor(basePath, sources)

  return descriptors.map(descriptor => {
    return {
      content: generate(meta, descriptor, options),
      path: descriptor.contentPath
    } as SFCFileInfo
  })
}

function generate (meta: MetaLocaleMessage, descriptor: SFCDescriptor, options?: FormatOptions): string {
  const i18nBlocks = meta.components[descriptor.contentPath]
  debug('target i18n blocks\n', i18nBlocks)

  const blocks: SFCBlock[] = getBlocks(descriptor)
  blocks.forEach(b => debug(`block: type=${b.type}, start=${b.start}, end=${b.end}`))

  const { raw } = descriptor
  const content = buildContent(i18nBlocks, raw, blocks, options)
  debug(`build content:\n${content}`)
  debug(`content size: raw=${raw.length}, content=${content.length}`)

  return content
}

function getBlocks (descriptor: SFCDescriptor): SFCBlock[] {
  const { template, script, styles, customBlocks } = descriptor
  const blocks: SFCBlock[] = [...styles, ...customBlocks]
  template && blocks.push(template as SFCBlock)
  script && blocks.push(script as SFCBlock)
  blocks.sort((a, b) => { return (a.start as number) - (b.start as number) })
  return blocks
}

function buildContent (i18nBlocks: SFCI18nBlock[], raw: string, blocks: SFCBlock[], options?: FormatOptions): string {
  let offset = 0
  let i18nBlockCounter = 0
  let contents: string[] = []

  contents = blocks.reduce((contents, block) => {
    if (block.type === 'i18n') {
      let lang = block.attrs.lang
      lang = (!lang || typeof lang !== 'string') ? 'json' : lang
      const locale: Locale | undefined = block.attrs.locale
      const i18nBlock = i18nBlocks[i18nBlockCounter]
      debug(`meta.lang = ${i18nBlock.lang}, block.lang = ${lang}, meta.locale = ${i18nBlock.locale}, block.locale = ${locale}`)

      let messages: any = null
      if (lang === i18nBlock.lang && locale === i18nBlock.locale) {
        if (locale) {
          messages = i18nBlock.messages[locale]
        } else {
          messages = i18nBlock.messages
        }
      } else {
        debug(`unmatch meta block and sfc block`)
        messages = parseContent(block.content, lang)
      }

      contents = contents.concat(raw.slice(offset, block.start))
      const serialized = `\n${stringifyContent(messages, lang, options)}`
      contents = contents.concat(serialized)
      offset = block.end as number
      i18nBlockCounter++
    } else {
      contents = contents.concat(raw.slice(offset, block.end))
      offset = block.end as number
    }
    return contents
  }, contents)
  contents = contents.concat(raw.slice(offset, raw.length))

  if (i18nBlocks.length > i18nBlockCounter) {
    i18nBlocks.slice(i18nBlockCounter).reduce((contents, i18nBlock) => {
      contents.push(buildI18nTag(i18nBlock, options))
      return contents
    }, contents)
  }

  return contents.join('')
}

function buildI18nTag (i18nBlock: SFCI18nBlock, options?: FormatOptions): string {
  const { locale, lang, messages } = i18nBlock
  let tag = '<i18n'
  if (locale) {
    tag += ` locale="${escape(locale)}"`
  }
  if (lang !== 'json') {
    tag += ` lang="${escape(lang)}"`
  }
  tag += '>'

  return `\n
${tag}
${stringifyContent(locale ? messages[locale] : messages, lang, options)}</i18n>`
}
