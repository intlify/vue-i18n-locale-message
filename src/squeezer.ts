import { SFCBlock } from 'vue-template-compiler'
import { MetaLocaleMessage, I18nLang, SFCFileInfo, SFCI18nBlock } from '../types'

import { reflectSFCDescriptor, parseContent } from './utils'

import { debug as Debug } from 'debug'
const debug = Debug('vue-i18n-locale-message:squeezer')

export default function squeeze (basePath: string, files: SFCFileInfo[]): MetaLocaleMessage {
  const descriptors = reflectSFCDescriptor(basePath, files)
  return descriptors.reduce((meta, descriptor) => {
    descriptor.customBlocks.sort((a, b) => { return (a.start as number) - (b.start as number) })
    const i18nBlocks = squeezeFromCustomBlocks(descriptor.customBlocks)
    debug('squeezeFromCustomBlocks: i18nBlocks', JSON.stringify(i18nBlocks, null, 2))
    meta.components[descriptor.contentPath] = i18nBlocks
    return meta
  }, { target: basePath, components: {}} as MetaLocaleMessage)
}

function squeezeFromCustomBlocks (blocks: SFCBlock[]): SFCI18nBlock[] {
  return blocks.map(block => {
    if (block.type === 'i18n') {
      debug('i18n block attrs', block.attrs)

      let lang = block.attrs.lang
      lang = (!lang || typeof lang !== 'string') ? 'json' : lang

      const i18nBlock: SFCI18nBlock = {
        lang: lang as I18nLang,
        messages: {}
      }
      const obj = parseContent(block.content, lang)

      const locale = block.attrs.locale
      if (!locale || typeof locale !== 'string') {
        Object.assign(i18nBlock.messages, obj)
      } else {
        i18nBlock.locale = locale
        Object.assign(i18nBlock.messages, { [locale]: obj })
      }

      return i18nBlock
    }
  }).filter(Boolean) as SFCI18nBlock[]
}
