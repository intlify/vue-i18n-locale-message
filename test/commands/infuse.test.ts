import * as yargs from 'yargs'
import deepmerge from 'deepmerge'
import path from 'path'

import jsonMetaInfo from '../fixtures/meta/json'
import json from '../fixtures/squeeze'
import external from '../fixtures/external'

// -------
// mocking

const TARGET_PATH = '/path/to/project1'
const SFC_FILES = [
  `${TARGET_PATH}/src/App.vue`,
  `${TARGET_PATH}/src/components/Modal.vue`,
  `${TARGET_PATH}/src/components/nest/RankingTable.vue`,
  `${TARGET_PATH}/src/pages/Login.vue`
]
const LOCALE_FILES = [
  `${TARGET_PATH}/src/locales/ja.json`,
  `${TARGET_PATH}/src/locales/en.json`
]
const MOCK_FILES = SFC_FILES.reduce((files, file) => {
  const meta = jsonMetaInfo.find(meta => meta.contentPath === file)
  return Object.assign(files, { [file]: meta.raw })
}, {})

let orgCwd // for process.cwd mock

// mock: ../../src/utils
jest.mock('../../src/utils', () => ({
  __esModule: true,
  ...jest.requireActual('../../src/utils'),
  loadNamespaceDictionary: jest.fn(),
  splitLocaleMessages: jest.fn(),
  resolve: jest.fn()
}))
import * as utils from '../../src/utils'

// mock: fs
jest.mock('fs', () => ({
  ...jest.requireActual('fs'),
  readFileSync: jest.fn(),
  writeFileSync: jest.fn()
}))
import fs from 'fs'

// mock: glob
jest.mock('glob', () => ({ sync: jest.fn((pattern) => {
  if (`${TARGET_PATH}/src/locales/*.json` === pattern) {
    return LOCALE_FILES
  } else {
    return SFC_FILES
  }
}) }))

// -------------------
// setup/teadown hooks

beforeEach(() => {
  orgCwd = process.cwd
  process.cwd = jest.fn(() => TARGET_PATH) // mock: process.cwd
})

afterEach(() => {
  jest.clearAllMocks()
  process.cwd = orgCwd
})

// ----------
// test cases

test('absolute path', async () => {
  // setup mocks
  const mockUtils = utils as jest.Mocked<typeof utils>
  mockUtils.resolve.mockImplementation((...paths) => paths[0])
  mockUtils.loadNamespaceDictionary.mockImplementation(async () => ({}))
  mockUtils.splitLocaleMessages.mockImplementation((messages) => ({ sfc: messages }))
  const writeFiles = {}
  const mockFS = fs as jest.Mocked<typeof fs>
  mockFS.readFileSync.mockImplementation(path => {
    if (MOCK_FILES[path as string]) {
      return MOCK_FILES[path as string]
    } else {
      return JSON.stringify(json)
    }
  })
  mockFS.writeFileSync.mockImplementation((path, data) => {
    writeFiles[path as string] = data.toString()
  })

  // run
  const infuse = await import('../../src/commands/infuse')
  const cmd = yargs.command(infuse)
  const output = await new Promise(resolve => {
    cmd.parse(`infuse --target=${TARGET_PATH}/src --locales=${TARGET_PATH}/locales.json`, () => {
      resolve(writeFiles)
    })
  })

  // check
  for (const [key, value] of Object.entries(output)) {
    expect(value).toMatchSnapshot(key)
  }
})

test('relative path', async () => {
  // setup mocks
  const mockUtils = utils as jest.Mocked<typeof utils>
  mockUtils.resolve
    .mockImplementationOnce(() => `${TARGET_PATH}/src`)
    .mockImplementationOnce((...paths) => `${TARGET_PATH}/${paths[0]}`)
  mockUtils.loadNamespaceDictionary.mockImplementation(async () => ({}))
  mockUtils.splitLocaleMessages.mockImplementation((messages) => ({ sfc: messages }))
  const writeFiles = {}
  const mockFS = fs as jest.Mocked<typeof fs>
  mockFS.readFileSync.mockImplementation(path => {
    if (MOCK_FILES[path as string]) {
      return MOCK_FILES[path as string]
    } else {
      return JSON.stringify(json)
    }
  })
  mockFS.writeFileSync.mockImplementation((path, data) => {
    writeFiles[path as string] = data.toString()
  })

  // run
  const infuse = await import('../../src/commands/infuse')
  const cmd = yargs.command(infuse)
  const output = await new Promise(resolve => {
    cmd.parse(`infuse --target=./src --locales=locales-2.json`, () => {
      resolve(writeFiles)
    })
  })

  // check
  for (const [key, value] of Object.entries(output)) {
    expect(value).toMatchSnapshot(key)
  }
})

test('dryRun option', async () => {
  // setup mock'
  const mockUtils = utils as jest.Mocked<typeof utils>
  mockUtils.resolve
    .mockImplementationOnce(() => `${TARGET_PATH}/src`)
    .mockImplementationOnce((...paths) => `${TARGET_PATH}/${paths[0]}`)
  mockUtils.loadNamespaceDictionary.mockImplementation(async () => ({}))
  mockUtils.splitLocaleMessages.mockImplementation((messages) => ({ sfc: messages }))
  const mockFS = fs as jest.Mocked<typeof fs>
  mockFS.readFileSync.mockImplementation(path => {
    if (MOCK_FILES[path as string]) {
      return MOCK_FILES[path as string]
    } else {
      return JSON.stringify(json)
    }
  })

  // run
  const infuse = await import('../../src/commands/infuse')
  const cmd = yargs.command(infuse)
  await new Promise(resolve => {
    cmd.parse(`infuse --target=./src --locales=locales-2.json --dry-run`, () => {
      resolve()
    })
  })

  // check
  expect(mockFS.writeFileSync).not.toHaveBeenCalled()
})

test('match option', async () => {
  // setup mocks
  const mockUtils = utils as jest.Mocked<typeof utils>
  mockUtils.resolve
    .mockImplementationOnce(() => `${TARGET_PATH}/src`)
    .mockImplementationOnce((...paths) => `${TARGET_PATH}/${paths[0]}`)
  mockUtils.loadNamespaceDictionary.mockImplementation(async () => ({}))
  mockUtils.splitLocaleMessages.mockImplementation((messages) => ({ sfc: messages }))
  const writeFiles = {}
  const mockFS = fs as jest.Mocked<typeof fs>
  mockFS.readFileSync.mockImplementation(p => {
    if (MOCK_FILES[p as string]) {
      return MOCK_FILES[p as string]
    } else {
      return JSON.stringify(json[path.basename(p as string, '.json')])
    }
  })
  mockFS.writeFileSync.mockImplementation((path, data) => {
    writeFiles[path as string] = data.toString()
  })

  // run
  const infuse = await import('../../src/commands/infuse')
  const cmd = yargs.command(infuse)
  const output = await new Promise(resolve => {
    cmd.parse(`infuse --target=./src --locales=./src/locales --match=^([\\w-]*)\\.json`, () => {
      resolve(writeFiles)
    })
  })

  // check
  for (const [key, value] of Object.entries(output)) {
    expect(value).toMatchSnapshot(key)
  }
})

test('bundle option', async () => {
  // setup mocks
  const mockUtils = utils as jest.Mocked<typeof utils>
  mockUtils.resolve
    .mockImplementationOnce(() => `${TARGET_PATH}/src`)
    .mockImplementationOnce((...paths) => `${TARGET_PATH}/${paths[0]}`)
  mockUtils.loadNamespaceDictionary.mockImplementation(async () => ({
    './test/fixtures/packages/package1/locales/**/*.json': 'package1',
    './test/fixtures/packages/package2/locales/**/*.json': 'package2'
  }))
  mockUtils.splitLocaleMessages.mockImplementation((messages) => ({
    sfc: deepmerge(messages, external),
    external: [{
      path: './test/fixtures/packages/package1/locales/en/common.json',
      messages: {
        package1: { common: { navigation: { title: 'title' }}}
      }
    }, {
      path: './test/fixtures/packages/package1/locales/ja/common.json',
      messages: {
        package1: { common: { navigation: { title: 'タイトル' }}}
      }
    }, {
      path: './test/fixtures/packages/package2/locales/en/profile.json',
      messages: {
        package2: {
          // eslint-disable-next-line @typescript-eslint/camelcase
          profile: { display_name: 'Display Name', email: 'E-Mail' }
        }
      }
    }, {
      path: './test/fixtures/packages/package2/locales/ja/profile.json',
      messages: {
        package2: { profile: {}}
      }
    }]
  }))
  const writeFiles = {}
  const mockFS = fs as jest.Mocked<typeof fs>
  mockFS.readFileSync.mockImplementation(p => {
    if (MOCK_FILES[p as string]) {
      return MOCK_FILES[p as string]
    } else {
      return JSON.stringify(json[path.basename(p as string, '.json')])
    }
  })
  mockFS.writeFileSync.mockImplementation((path, data) => {
    writeFiles[path as string] = data.toString()
  })

  // run
  const infuse = await import('../../src/commands/infuse')
  const cmd = yargs.command(infuse)
  const output = await new Promise(resolve => {
    cmd.parse(`infuse --target=./src --locales=./src/locales --match=^([\\w-]*)\\.json \
      --unbundleTo=./test/fixtures/packages/package1/locales/**/*.json,./test/fixtures/packages/package2/locales/**/*.json \
      --unbundleMatch=([\\w]*)/([\\w]*)\\.json$ \
      --namespace=./test/fixtures/namespace.json`, () => {
      resolve(writeFiles)
    })
  })

  expect(output).toMatchSnapshot()
})
