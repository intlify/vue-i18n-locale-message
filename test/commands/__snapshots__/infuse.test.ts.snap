// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`absolute path: /path/to/project1/src/App.vue 1`] = `
"
  <template>
    <p>template</p>
  </template>
  
  <script>
  export default {}
  </script>

  <i18n>
{
  \\"en\\": {
    \\"title\\": \\"Application\\"
  },
  \\"ja\\": {
    \\"title\\": \\"アプリケーション\\"
  }
}
</i18n>
  "
`;

exports[`absolute path: /path/to/project1/src/components/Modal.vue 1`] = `
"
  <template>
    <p>template</p>
  </template>
  
  <script>
  export default {}
  </script>

  <i18n locale=\\"en\\">
{
  \\"ok\\": \\"OK\\",
  \\"cancel\\": \\"Cancel\\"
}
</i18n>
  
  <i18n locale=\\"ja\\">
{
  \\"ok\\": \\"OK\\",
  \\"cancel\\": \\"キャンセル\\"
}
</i18n>
  "
`;

exports[`absolute path: /path/to/project1/src/components/nest/RankingTable.vue 1`] = `
"
  <template>
    <p>template</p>
  </template>
  
  <script>
  export default {}
  </script>

  <i18n locale=\\"en\\">
{
  \\"headers\\": {
    \\"rank\\": \\"Rank\\",
    \\"name\\": \\"Name\\",
    \\"score\\": \\"Score\\"
  }
}
</i18n>
  

<i18n locale=\\"ja\\">
{
  \\"headers\\": {
    \\"rank\\": \\"ランク\\",
    \\"name\\": \\"名前\\",
    \\"score\\": \\"スコア\\"
  }
}
</i18n>"
`;

exports[`absolute path: /path/to/project1/src/pages/Login.vue 1`] = `
"
  <template>
    <p>template</p>
  </template>
  
  <script>
  export default {}
  </script>

  <i18n>
{
  \\"ja\\": {
    \\"id\\": \\"ユーザーID\\",
    \\"passowrd\\": \\"パスワード\\",
    \\"confirm\\": \\"パスワードの確認入力\\",
    \\"button\\": \\"ログイン\\"
  }
}
</i18n>
  

<i18n locale=\\"en\\">
{
  \\"id\\": \\"User ID\\",
  \\"password\\": \\"Password\\",
  \\"confirm\\": \\"Confirm Password\\",
  \\"button\\": \\"Login\\"
}
</i18n>"
`;

exports[`bundle option 1`] = `
Object {
  "./test/fixtures/packages/package1/locales/en/common.json": "{
  \\"package1\\": {
    \\"common\\": {
      \\"navigation\\": {
        \\"title\\": \\"title\\"
      }
    }
  }
}",
  "./test/fixtures/packages/package1/locales/ja/common.json": "{
  \\"package1\\": {
    \\"common\\": {
      \\"navigation\\": {
        \\"title\\": \\"タイトル\\"
      }
    }
  }
}",
  "./test/fixtures/packages/package2/locales/en/profile.json": "{
  \\"package2\\": {
    \\"profile\\": {
      \\"display_name\\": \\"Display Name\\",
      \\"email\\": \\"E-Mail\\"
    }
  }
}",
  "./test/fixtures/packages/package2/locales/ja/profile.json": "{
  \\"package2\\": {
    \\"profile\\": {}
  }
}",
  "/path/to/project1/src/App.vue": "
  <template>
    <p>template</p>
  </template>
  
  <script>
  export default {}
  </script>

  <i18n>
{
  \\"en\\": {
    \\"title\\": \\"Application\\"
  },
  \\"ja\\": {
    \\"title\\": \\"アプリケーション\\"
  }
}
</i18n>
  ",
  "/path/to/project1/src/components/Modal.vue": "
  <template>
    <p>template</p>
  </template>
  
  <script>
  export default {}
  </script>

  <i18n locale=\\"en\\">
{
  \\"ok\\": \\"OK\\",
  \\"cancel\\": \\"Cancel\\"
}
</i18n>
  
  <i18n locale=\\"ja\\">
{
  \\"ok\\": \\"OK\\",
  \\"cancel\\": \\"キャンセル\\"
}
</i18n>
  ",
  "/path/to/project1/src/components/nest/RankingTable.vue": "
  <template>
    <p>template</p>
  </template>
  
  <script>
  export default {}
  </script>

  <i18n locale=\\"en\\">
{
  \\"headers\\": {
    \\"rank\\": \\"Rank\\",
    \\"name\\": \\"Name\\",
    \\"score\\": \\"Score\\"
  }
}
</i18n>
  

<i18n locale=\\"ja\\">
{
  \\"headers\\": {
    \\"rank\\": \\"ランク\\",
    \\"name\\": \\"名前\\",
    \\"score\\": \\"スコア\\"
  }
}
</i18n>",
  "/path/to/project1/src/pages/Login.vue": "
  <template>
    <p>template</p>
  </template>
  
  <script>
  export default {}
  </script>

  <i18n>
{
  \\"ja\\": {
    \\"id\\": \\"ユーザーID\\",
    \\"passowrd\\": \\"パスワード\\",
    \\"confirm\\": \\"パスワードの確認入力\\",
    \\"button\\": \\"ログイン\\"
  }
}
</i18n>
  

<i18n locale=\\"en\\">
{
  \\"id\\": \\"User ID\\",
  \\"password\\": \\"Password\\",
  \\"confirm\\": \\"Confirm Password\\",
  \\"button\\": \\"Login\\"
}
</i18n>",
}
`;

exports[`match option: /path/to/project1/src/App.vue 1`] = `
"
  <template>
    <p>template</p>
  </template>
  
  <script>
  export default {}
  </script>

  <i18n>
{
  \\"en\\": {
    \\"title\\": \\"Application\\"
  },
  \\"ja\\": {
    \\"title\\": \\"アプリケーション\\"
  }
}
</i18n>
  "
`;

exports[`match option: /path/to/project1/src/components/Modal.vue 1`] = `
"
  <template>
    <p>template</p>
  </template>
  
  <script>
  export default {}
  </script>

  <i18n locale=\\"en\\">
{
  \\"ok\\": \\"OK\\",
  \\"cancel\\": \\"Cancel\\"
}
</i18n>
  
  <i18n locale=\\"ja\\">
{
  \\"ok\\": \\"OK\\",
  \\"cancel\\": \\"キャンセル\\"
}
</i18n>
  "
`;

exports[`match option: /path/to/project1/src/components/nest/RankingTable.vue 1`] = `
"
  <template>
    <p>template</p>
  </template>
  
  <script>
  export default {}
  </script>

  <i18n locale=\\"en\\">
{
  \\"headers\\": {
    \\"rank\\": \\"Rank\\",
    \\"name\\": \\"Name\\",
    \\"score\\": \\"Score\\"
  }
}
</i18n>
  

<i18n locale=\\"ja\\">
{
  \\"headers\\": {
    \\"rank\\": \\"ランク\\",
    \\"name\\": \\"名前\\",
    \\"score\\": \\"スコア\\"
  }
}
</i18n>"
`;

exports[`match option: /path/to/project1/src/pages/Login.vue 1`] = `
"
  <template>
    <p>template</p>
  </template>
  
  <script>
  export default {}
  </script>

  <i18n>
{
  \\"ja\\": {
    \\"id\\": \\"ユーザーID\\",
    \\"passowrd\\": \\"パスワード\\",
    \\"confirm\\": \\"パスワードの確認入力\\",
    \\"button\\": \\"ログイン\\"
  }
}
</i18n>
  

<i18n locale=\\"en\\">
{
  \\"id\\": \\"User ID\\",
  \\"password\\": \\"Password\\",
  \\"confirm\\": \\"Confirm Password\\",
  \\"button\\": \\"Login\\"
}
</i18n>"
`;

exports[`relative path: /path/to/project1/src/App.vue 1`] = `
"
  <template>
    <p>template</p>
  </template>
  
  <script>
  export default {}
  </script>

  <i18n>
{
  \\"en\\": {
    \\"title\\": \\"Application\\"
  },
  \\"ja\\": {
    \\"title\\": \\"アプリケーション\\"
  }
}
</i18n>
  "
`;

exports[`relative path: /path/to/project1/src/components/Modal.vue 1`] = `
"
  <template>
    <p>template</p>
  </template>
  
  <script>
  export default {}
  </script>

  <i18n locale=\\"en\\">
{
  \\"ok\\": \\"OK\\",
  \\"cancel\\": \\"Cancel\\"
}
</i18n>
  
  <i18n locale=\\"ja\\">
{
  \\"ok\\": \\"OK\\",
  \\"cancel\\": \\"キャンセル\\"
}
</i18n>
  "
`;

exports[`relative path: /path/to/project1/src/components/nest/RankingTable.vue 1`] = `
"
  <template>
    <p>template</p>
  </template>
  
  <script>
  export default {}
  </script>

  <i18n locale=\\"en\\">
{
  \\"headers\\": {
    \\"rank\\": \\"Rank\\",
    \\"name\\": \\"Name\\",
    \\"score\\": \\"Score\\"
  }
}
</i18n>
  

<i18n locale=\\"ja\\">
{
  \\"headers\\": {
    \\"rank\\": \\"ランク\\",
    \\"name\\": \\"名前\\",
    \\"score\\": \\"スコア\\"
  }
}
</i18n>"
`;

exports[`relative path: /path/to/project1/src/pages/Login.vue 1`] = `
"
  <template>
    <p>template</p>
  </template>
  
  <script>
  export default {}
  </script>

  <i18n>
{
  \\"ja\\": {
    \\"id\\": \\"ユーザーID\\",
    \\"passowrd\\": \\"パスワード\\",
    \\"confirm\\": \\"パスワードの確認入力\\",
    \\"button\\": \\"ログイン\\"
  }
}
</i18n>
  

<i18n locale=\\"en\\">
{
  \\"id\\": \\"User ID\\",
  \\"password\\": \\"Password\\",
  \\"confirm\\": \\"Confirm Password\\",
  \\"button\\": \\"Login\\"
}
</i18n>"
`;
